src/sas/sascalc/calculator/gsc_model.py

```diff
@@ -0,0 +1,156 @@
+"""
+create plugin model from the Generic Scattering Calculator
+"""
+import os
+import logging
+import math
+
+from sas.sascalc.fit import models
+from sas.qtgui.Utilities.TabbedModelEditor import TabbedModelEditor
+
+
+def writePlugin(self):
+    # check if file exists & assign filename
+    plugin_location = models.find_plugins_dir()
+    full_path = os.path.join(plugin_location, self.txtFileName.text())
+    if os.path.splitext(full_path)[1] != ".py":
+        full_path += ".py"
+
+    # generate the model representation as string
+    model_str = generateModel(self)
+    TabbedModelEditor.writeFile(full_path, model_str)
+
+    self.manager.communicate.customModelDirectoryChanged.emit()
```

Comment:
Importing qtgui from sascalc should never happen.

writeFile: Return the model string from here and use the TabbedModelEditor file writer from there -or- port the write method to here since it's pretty boilerplate.

self.manger doesn't exist in this context so this needs to move back regardless.

Reply:
Has this been resolved?